walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,52 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashSet;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.logging.Logger;
 
+//changed from public final class to private static
 public final class FindMeetingQuery {
+  private static Logger log = Logger.getLogger("FindMeetingQuery");
+
+  //will use these later for manual testing
+  private static final String PERSON_A = "Person A";
+  private static final String PERSON_B = "Person B";
+  private static final String PERSON_C = "Person C";
+
+  private static final int TIME_0800AM = TimeRange.getTimeInMinutes(8, 0);
+  private static final int TIME_0830AM = TimeRange.getTimeInMinutes(8, 30);
+  private static final int TIME_0900AM = TimeRange.getTimeInMinutes(9, 0);
+  private static final int TIME_0930AM = TimeRange.getTimeInMinutes(9, 30);
+  private static final int TIME_1000AM = TimeRange.getTimeInMinutes(10, 0);
+  private static final int TIME_1100AM = TimeRange.getTimeInMinutes(11, 00);
+
+  private static final int DURATION_30_MINUTES = 30;
+  private static final int DURATION_60_MINUTES = 60;
+  private static final int DURATION_90_MINUTES = 90;
+  private static final int DURATION_1_HOUR = 60;
+  private static final int DURATION_2_HOUR = 120;
+
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+
+    //Collections.sort() only takes in List, so must convert events to a list
+    List<Event> eventsAsList = new ArrayList<Event>();
+    for(Event event: events){
+      eventsAsList.add(event);
+    }
+
+    Collections.sort(eventsAsList, Event.ORDER_BY_START);
```

Comment:
Alternatively you can also convert the Event to a TimeRange, and just use 

`Collections.sort(eventsAsRanges, TimeRange.ORDER_BY_START);` instead of defining a new comparator in Event

Reply:
Specifically Event.getWhen() returns a TimeRange corresponding to the event

Reply:
Why did I not think of that, that makes a lot more sense! Note to self: look at what functions I'm given & realize that they are given to me for a reason (whew)

Reply:
Lol no worries! We all do this. The project description is super vague.